torch/hub.py

```diff
@@ -210,6 +216,71 @@ def _get_cache_or_reload(github, force_reload, verbose=True, skip_validation=Fal
     return repo_dir
 
 
+
+def _add_repo_to_trusted_list(repo, filepath, is_trusted=False):
+    # prompt
+    response = input(
+        f"The repository {repo} does not belong to the list of trusted repositories and as such cannot be downloaded. "
+        "Do you trust this repository and wish to add it to the trusted list of repositories (Y/n)?")
+    if response.lower() in {"y", 'yes'}:
+        if is_trusted:
+            print("The repository is already trusted.")
+            return
+        else:
+            with open(filepath, "a") as file:
+                file.write(repo + "\n")
+
+    elif response.lower() in {"n", "no"}:
+        raise Exception("Untrusted repository.")
+
+    else:
+        raise ValueError(f"Unrecognized response {response}.")
+
+
+def _check_repo(repo, trust_repo=None, calling_fn="load"):
+    hub_dir = get_dir()
+    if not os.path.exists(hub_dir):
+        os.makedirs(hub_dir)
+    filepath = os.path.join(hub_dir, "trusted_list")
+
+    if not os.path.exists(filepath):
+        Path(filepath).touch()
+
+        # Initialize with all existing repos
+        repos = next(os.walk(hub_dir))[1]
+        with open(filepath, "a") as file:
+            for _repo in repos:
+                file.write(_repo + "\n")
+
+    # load list
+    is_trusted = any([repo.startswith(_predefined_trusted) for _predefined_trusted in PREDEFINED_TRUSTED])
+    with open(filepath, 'r') as file:
+        for _repo in file:
+            is_trusted = is_trusted or _repo.startswith(repo)
+            if is_trusted:
+                break
```

Comment:
Would something like this be simpler?

```py
trusted_repos = _PREDEFINED_TRUSTED
with open(filepath, 'r') as file:
    trusted_repos += file.readlines()
is_trusted = any(repo.startswith(trusted_repo) for trusted_repo in trusted_repos)
```

Reply:
sure but with a copy of `_PREDEFINED_TRUSTED`:
```
trusted_repos = [repo for repo in _PREDEFINED_TRUSTED]
```
otherwise if one calls `torch.hub.load` twice in the same session the global list will be incremented twice with the same elements.